### ISSUE TITLE
[BUG] Can't run playwright docker image with volumes on windows

### ISSUE THREAD

**Issue by Brianzchen** [microsoft/playwright#26735]
### System info
- Playwright Version: 1.37.1
- Operating System: Windows 11
- Browser: All
- Other info: With powershell, no WSL2

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
const { defineConfig, devices } = require('@playwright/test');

/**
 * @see https://playwright.dev/docs/test-configuration
 */
module.exports = (defineConfig({
  testMatch: 'src/**/vr/*.vr.js',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [
    // We always run this in docker so we don't want the docker image to hang
    ['html', { open: 'never' }],
  ],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://127.0.0.1:3000',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
  },
  webServer: {
    command: 'vr=true headless=true yarn website:start',
    url: 'http://127.0.0.1:3000',
    reuseExistingServer: !process.env.CI,
    stdout: 'ignore',
    stderr: 'pipe',
  },
  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },

    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },
    },

    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: { ...devices['Pixel 5'] },
    // },
    {
      name: 'Mobile Safari',
      use: {
        ...devices['iPhone 12'],
      },
    },

    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
    // },
  ],
}));
```

**Test file (self-contained)**

Any test file should be fine

```js
it('should check the box using setChecked', async ({ page }) => {
  await page.setContent(`<input id='checkbox' type='checkbox'></input>`);
  await page.getByRole('checkbox').check();
  await expect(page.getByRole('checkbox')).toBeChecked();
});
```

**Steps**
I have a very simple docker image

```
FROM mcr.microsoft.com/playwright:v1.37.1-jammy

ADD . /boba
WORKDIR /boba

CMD ["yarn", "test:vr-ci"]
```

where `test:vr-ci` is `playwright test`

That I am building to run on with unix based systems, but can also be called from powershell as my team is OS agnostic. The unix side works fine but on powershell, git bash or anything when running on a windows system times out and fails whenever I mount a volume to be able to write data back into the repo such as screenshots and test report.

I have ensured that I have enabled file sharing for docker and when I run a command on powershell like `docker run -v ${PWD}:/data alpine rm /data/yarn.lock` it works perfectly fine so I don't think it's volume sharing or permission related, but when I change that to `docker run -v ${PWD}:/boba playwright` where playwright is my docker build name, it hangs and then starts the web server to promptly timeout.

![Screenshot 2023-08-28 213019](https://github.com/microsoft/playwright/assets/12436524/c64766cd-6195-495a-b9ba-ce3c125bb913)

Also to note `docker run playwright` works fine, it's only when I mount a volume

**Comment by mxschmitt** [comment]
Would it be possible to share a reproduction in form of a GitHub repository with us? Hanging sounds definitely not expected.

**Comment by Brianzchen** [comment]
I just made a repro but surprisingly I was actually able to get tests with screenshots working on windows git bash. I'll have to work backwards more to find which part causes it to die

**Comment by aslushnikov** [comment]
I'll close this then; feel free to file a new issue once you have a good repro that demonstrates that it's Playwright fault!
